### PR TITLE
scylla-gdb.py: use chunked_fifo to represent _sink._pending_io

### PR DESCRIPTION
**Pull request by tchaikov**
we switched from `circular_buffer` to `chunked_fifo` to present `io_sink::_pending_io` in the latest seastar now. to be prepared for this change, let's

* add `chunked_fifo` class in `scylla-gdb.py`.
* use `circular_buffer` as a fallback of `chunked_fifo`. instead of doing this the other way around, we try to send the message that the latest seastar uses `chunked_fifo`.

---

this change prepares for a succeeding change to bump up the seastar  module, which does not include critical fixes, so we don't need to backport this change.

### CODE REVIEW

**Comment by tchaikov** [comment]
see also https://github.com/scylladb/seastar/commit/3859cf423051d11a39d280041700ed07aeba6272  and the [PR](https://github.com/scylladb/seastar/pull/2357) introducing this change

**Comment by tchaikov** [comment]
@xemul hi Pavel, since you reviewed https://github.com/scylladb/seastar/pull/2357 , and has been working on the io scheduler of seastar, could you help review this change as well?

**Comment by tchaikov** [comment]
tested at #20281

**Comment by tchaikov** [comment]
> #### 🔴 CI State: FAILURE
> 
> ✅ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11091/) ✅ - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11091/) ✅ - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/3187/) ❌ - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/11211/) ✅ - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/141/) ✅ - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11091/)
> #### Build Details:
> 
>     * Duration: 3 hr 8 min
> 
>     * Builder: spider2.cloudius-systems.com

```
jenkins pipeline failure: Error when executing always post condition:

hudson.AbortException: Spot termination
```

the failure was caused by infra issues.

File: scylla-gdb.py [hunk 1]
```diff
@@ -3789,7 +3854,11 @@ def invoke(self, arg, for_tty):
                 for pclass_ptr in handles:
                     self._print_io_priority_class(fq_pclass(pclass_ptr), names_from_ptrs)
 
-            pending = circular_buffer(ioq['_sink']['_pending_io'])
+            try:
+                pending = chunked_fifo(ioq['_sink']['_pending_io'])
+                _ = len(pending)
```
Comment (xemul) [inline scylla-gdb.py:3859]: Why is this needed? Presumably that's the place that trows in case pending-io is not chunked fifo

Comment (tchaikov) [reply scylla-gdb.py:3859]: exactly. it's not necessary anymore. it was added for 
```py
try:
    pending = circular_buffer(ioq['_sink']['_pending_io'])
except gdb.error:
    # ..
```
as previously the constructor of `circular_buffer` does not throw. but since i put `chunked_info` in the `try` block, it does throw if the inspected variable's type does not match.

so, it's a leftover. will drop it.

**Comment by tchaikov** [comment]
v2:

- removed the extraneous `_ = len(pending)`

@xemul hi Pavel, could you take another look?

File: scylla-gdb.py (hunk 1 above)
Comment (xemul) [reply scylla-gdb.py]: Here's how it is initialized now

```python
            try:
                pending = chunked_fifo(ioq['_sink']['_pending_io'])
            except gdb.error:
                pending = circular_buffer(ioq['_sink']['_pending_io'])
```

so it attempts to set pending as chunked fifo and falls back to circular buffer if it throws. But chunked fifo constructor is just

```python
    def __init__(self, ref):
        self._ref = ref
```

it doesn't do any type-checking, so why would it throw?

Comment (tchaikov) [reply scylla-gdb.py]: sorry, i missed this. now i added a 
```py
_ = self.front_chunk
```
in the ctor. should be better now.

**Comment by tchaikov** [comment]
v3:

- try to read the member variable , so that the constructor of `chunked_fifo` throws when it is applied to a variable with the wrong type.

**Review by xemul — APPROVED** [review]
Looks good, thanks!

**Comment by tchaikov** [comment]
@scylladb/scylla-maint hello maintainers, could you help queue this change?

**Review by denesb — APPROVED** [review]
